src/types.rs

```diff
@@ -0,0 +1,35 @@
+//! Concrete node and edge types used in the registry.
+
+/// The type of a node.
+pub enum NodeType {
+    /// A user, eg. contributor, project member etc.
+    User,
+    /// A project with users as members and contributors.
+    Project,
+}
+
+/// The type of an edge.
+pub enum EdgeType {
+    /// Contribution, user -> project, with strength metric.
+    ContributionToProject(u32),
+    /// Contribution, project -> user, with strength metric.
+    ContributionFromUser(u32),
+    /// Membership of a user in a project, project <-> user.
+    ProjectMembership,
+    /// One-way dependency between two projects, project -> project.
+    ProjectDependency,
+}
+
+/// The rank or "osrank" of a node, normalized to `1.0`.
+pub type NodeRank = f64;
+
+/// Global parameters used by the graph algorithm.
+pub struct HyperParameters {
+    /// Threshold below which nodes are pruned in the first
+    /// phase of the algorithm.
+    pruning_threshold: f64,
+    /// How often do random walks return to the seed set.
+    dampening_factor: f64,
```

Comment:
```suggestion
    damping_factor: f64,
```

There are actually 2 `damping_factor`s, one for projects and accounts. The comment suggests these are used for the seed set phase only but in fact they are used in both phases. So `...return to the starting nodes` would be a more correct explanation.

Reply:
Aren't the starting nodes the same for both phases?

Reply:
Ah no they are not, according to the paper.

Reply:
Are the damping factors applied on nodes or edges? If there are two damping factors but 4 edge types, how does that work?

Reply:
The damping factors are applied on nodes ie if the random walker is on a specific node the walk continues with probability of the damping factor of the type of the node. For more info check out section 4.2.1 of the [osrank model documentation](https://github.com/oscoin/docs/blob/master/osrank/spec/osrank_spec.pdf)

Reply:
Thanks, fixed.